code/src/sixsq/nuvla/server/resources/infrastructure_service_kubernetes.clj

```diff
@@ -43,27 +42,54 @@
 ;;
 
 (defmethod infra-service/post-add-hook method
-           [service request]
-           (try
-             (let [id (:id service)
-                   user-id (:identity (a/current-authentication request))
-                   {{job-id     :resource-id
-                     job-status :status} :body} (job/create-job id "start_infrastructure_service_kubernetes"
-                                                                {:owner {:principal "ADMIN"
-                                                                         :type      "ROLE"}
-                                                                 :rules [{:principal user-id
-                                                                          :right     "VIEW"
-                                                                          :type      "USER"}]}
-                                                                :priority 50)
-                   job-msg (str "starting " id " with async " job-id)]
-                  (when (not= job-status 201)
-                        (throw (r/ex-response "unable to create async job to start infrastructure service kubernetes" 500 id)))
-                  (-> id
-                      (db/retrieve request)
-                      (a/can-modify? request)
-                      (assoc :state "STARTING")
-                      (db/edit request))
-                  (event-utils/create-event id job-msg (a/default-acl (a/current-authentication request)))
-                  (r/map-response job-msg 202 id job-id))
-             (catch Exception e
-               (or (ex-data e) (throw e)))))
+  [service request]
+  (try
+    (let [id (:id service)
+          user-id (:identity (a/current-authentication request))
+          {{job-id     :resource-id
+            job-status :status} :body} (job/create-job id "start_infrastructure_service_kubernetes"
+                                                       {:owner {:principal "ADMIN"
+                                                                :type      "ROLE"}
+                                                        :rules [{:principal user-id
+                                                                 :right     "VIEW"
+                                                                 :type      "USER"}]}
+                                                       :priority 50)
+          job-msg (str "starting " id " with async " job-id)]
+      (when (not= job-status 201)
+        (throw (r/ex-response "unable to create async job to start infrastructure service kubernetes" 500 id)))
+      (-> id
+          (db/retrieve request)
+          (a/can-modify? request)
```

Comment:
This occurs in the wrong order.  You've created a job to start the service before checking that the user has the right to do so.

Reply:
hum, but how is this different from deployment.clj?

Reply:
If it's that way on the deployment, then it should be changed there as well.

Reply:
fixed

Reply:
fixed